### ISSUE TITLE
Feature Request: Actions: Support externally defined configuration files

### ISSUE THREAD

**Issue by kbingham** [go-gitea/gitea#24454]
Splitting out from the now closed #13539

> Yup now I've been exploring more, I can see a separated repo for the workflows could be a very helpful way of doing this too. Then multiple repos (mirrors, forks, or separate repos for separate developers) can point to common workflow actions.

FWIW - Today I discovered gitlab can do this and it's all documented here: 
 - https://docs.gitlab.com/ee/ci/pipelines/settings.html#specify-a-custom-cicd-configuration-file
 - https://gitlab.com/gitlab-org/gitlab/-/issues/14376

_Originally posted by @kbingham in https://github.com/go-gitea/gitea/issues/13539#issuecomment-1518219043_

**Comment by wolfogre** [comment]
I think [reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) can do the same thing.

**Comment by kbingham** [comment]
The problem I can see with reusable workflows is : https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow

The use case here means I can not add any specific CI files to the repository itself which is to be compiled. Specifically this is to support building libcamera and the linux kernel. 

I can't merge my CI files into the linux kernel, and I'm not allowed to merge my CI files into libcamera.

**Comment by Hypnotist1148** [comment]
> The problem I can see with reusable workflows is : https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow
> 
> The use case here means I can not add any specific CI files to the repository itself which is to be compiled. Specifically this is to support building libcamera and the linux kernel.
> 
> I can't merge my CI files into the linux kernel, and I'm not allowed to merge my CI files into libcamera.

Agree, being able to select common/shared configuration in the existing actions setting for a repository or even a whole organization would be a huge plus, especially for environments where Organizations and Repositories are managed by automated systems. Or even just small teams.

**Comment by sthomen** [comment]
I for one would very much prefer to not have to have _configuration of the server_ leak into my code repositories.

**Comment by silverwind** [comment]
Scoped workflows now allow centrally maintained Actions workflows to run for consuming repositories, including pull mirrors, without a workflow file in each repository.

Fixed in https://github.com/go-gitea/gitea/commit/f46c9a97698f4a9eac8ac7b16679c4949a271d43.

Written by Codex.
